From a3099488f793f73d655b43554298f185b269d960 Mon Sep 17 00:00:00 2001
From: Rui Teng <rui.teng@amd.com>
Date: Fri, 11 Oct 2019 17:53:16 +0800
Subject: [PATCH 4061/4256] drm/amd/display: add config check for DSC support
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fix dkms install issue:
‘struct dc_crtc_timing_flags’ has no member named ‘DSC’
‘struct dc_crtc_timing’ has no member named ‘dsc_cfg’

Signed-off-by: Rui Teng <rui.teng@amd.com>
Reviewed-by: changzhu <Changfeng.Zhu@amd.com>
---
 drivers/gpu/drm/amd/display/dc/dcn20/dcn20_resource.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/gpu/drm/amd/display/dc/dcn20/dcn20_resource.c b/drivers/gpu/drm/amd/display/dc/dcn20/dcn20_resource.c
index 5deb8deece0c..4ca819c223bd 100644
--- a/drivers/gpu/drm/amd/display/dc/dcn20/dcn20_resource.c
+++ b/drivers/gpu/drm/amd/display/dc/dcn20/dcn20_resource.c
@@ -1937,8 +1937,10 @@ int dcn20_populate_dml_pipes_from_context(
 			pipes[pipe_cnt].dout.output_bpp = output_bpc * 3;
 		}
 
+#ifdef CONFIG_DRM_AMD_DC_DSC_SUPPORT
 		if (res_ctx->pipe_ctx[i].stream->timing.flags.DSC)
 			pipes[pipe_cnt].dout.output_bpp = res_ctx->pipe_ctx[i].stream->timing.dsc_cfg.bits_per_pixel / 16.0;
+#endif
 
 		/* todo: default max for now, until there is logic reflecting this in dc*/
 		pipes[pipe_cnt].dout.output_bpc = 12;
-- 
2.17.1

